app/src/main/java/org/oppia/app/home/HomeFragmentPresenter.kt

```diff
@@ -148,4 +195,15 @@ class HomeFragmentPresenter @Inject constructor(
   fun onTopicSummaryClicked(topicSummary: TopicSummary) {
     routeToTopicListener.routeToTopic(topicSummary.topicId)
   }
+
+  // TODO(#555): Create one central utility file from where we should access date format or even convert date timestamp to string from that file.
+  private fun displayGreeting() {
+    val c = Calendar.getInstance()
+    when (c.get(Calendar.HOUR_OF_DAY)) {
+      in 5..11 -> userAppHistoryViewModel.greeting = fragment.requireContext().getString(R.string.good_morning)
+      in 12..16 -> userAppHistoryViewModel.greeting = fragment.requireContext().getString(R.string.good_afternoon)
+      in 17..22 -> userAppHistoryViewModel.greeting = fragment.requireContext().getString(R.string.good_evening)
+      in 23 downTo 4 -> userAppHistoryViewModel.greeting = fragment.requireContext().getString(R.string.good_evening)
```

Comment:
Last two conditions should be merged such that `in  17 downTo 4`

Reply:
updated.